src/main/java/uk/gov/hmcts/dts/fact/mapit/MapitValidator.java

```diff
@@ -0,0 +1,110 @@
+package uk.gov.hmcts.dts.fact.mapit;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.dts.fact.services.MapitService;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Slf4j
+@Component
+public class MapitValidator {
+
+    private final MapitService mapitService;
+    private static final String FULL_POSTCODE = "([Gg][Ii][Rr] 0[Aa]{2})|((([A-Za-z][0-9]{1,2})|(([A-Za-z]"
+        + "[A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y]"
+        + "[0-9][A-Za-z]?))))\\s?[0-9][A-Za-z]{2})";
+    private static final String PARTIAL_POSTCODE_SEARCH = "[A-Za-z]{2}";
+    private static final String PARTIAL_POSTCODE_NUMERIC = "(^[A-Za-z]{1,2}([0-9]{1,3}))$";
```

Comment:
There are 2 sets of brackets here. Do we need any of them? There is no AND or OR here so the regex should just flow from the start till end

Reply:
Think it stores it in a group, which is needed for the matcher below in the code to retrieve the value :)